src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SDSEarlyReleaseDefaultingRulesService.kt

```diff
@@ -1,70 +1,66 @@
 import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.enumerations.ReleaseDateType
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.enumerations.SDSEarlyReleaseTranche
 import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.enumerations.SentenceIdentificationTrack
 import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.Booking
 import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.CalculationResult
 import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.ReleaseDateCalculationBreakdown
 import java.time.LocalDate
 
 @Service
-class SDSEarlyReleaseDefaultingRulesService(
-  @Value("\${sds-early-release-tranches.tranche-one-date}")
-  @DateTimeFormat(pattern = "yyyy-MM-dd")
-  private val trancheOneCommencementDate: LocalDate,
-) {
+class SDSEarlyReleaseDefaultingRulesService {
 
-  fun requiresRecalculation(booking: Booking, result: CalculationResult): Boolean {
-    return hasAnySDSEarlyRelease(booking) && hasAnyReleaseBeforeTrancheCommencement(result)
+  fun requiresRecalculation(booking: Booking, result: CalculationResult, trancheCommencementDate: LocalDate?): Boolean {
+    return hasAnySDSEarlyRelease(booking) && hasAnyReleaseBeforeTrancheCommencement(result, trancheCommencementDate)
   }
 
-  fun mergeResults(earlyReleaseResult: CalculationResult, standardReleaseResult: CalculationResult): CalculationResult {
+  fun mergeResults(earlyReleaseResult: CalculationResult, standardReleaseResult: CalculationResult, trancheCommencementDate: LocalDate?, allocatedTranche: SDSEarlyReleaseTranche): CalculationResult {
     val dates = earlyReleaseResult.dates.toMutableMap()
     val breakdownByReleaseDateType = earlyReleaseResult.breakdownByReleaseDateType.toMutableMap()
 
     DATE_TYPES_TO_ADJUST_TO_COMMENCEMENT_DATE.forEach { releaseDateType ->
-      mergeDate(releaseDateType, earlyReleaseResult, standardReleaseResult, trancheOneCommencementDate, dates, breakdownByReleaseDateType)
+      mergeDate(releaseDateType, earlyReleaseResult, standardReleaseResult, trancheCommencementDate, dates, breakdownByReleaseDateType)
     }
 
     return CalculationResult(
       dates,
       breakdownByReleaseDateType,
       earlyReleaseResult.otherDates,
       earlyReleaseResult.effectiveSentenceLength,
+      sdsEarlyReleaseTranche = allocatedTranche,
     )
   }
 
   private fun hasAnySDSEarlyRelease(anInitialCalc: Booking) =
     anInitialCalc.sentences.any { it.identificationTrack == SentenceIdentificationTrack.SDS_EARLY_RELEASE }
 
-  private fun hasAnyReleaseBeforeTrancheCommencement(result: CalculationResult): Boolean {
+  private fun hasAnyReleaseBeforeTrancheCommencement(result: CalculationResult, trancheCommencementDate: LocalDate?): Boolean {
     return DATE_TYPES_TO_ADJUST_TO_COMMENCEMENT_DATE
       .mapNotNull { result.dates[it] }
-      .any { it < trancheOneCommencementDate }
+      .any { it < trancheCommencementDate }
```

Comment:
What happens if the commencement date is null here?

Reply:
It shouldn't cause a problem because of the `hasAnySDSEarlyRelease(booking)` on line 17 - it'll only ever be null if there isn't an SDS for early release, so that part of the IF won't be evaluated